### DISCUSSION TITLE
Suggested scalable deployment of nextjs? (multiple nextjs instances)

### DISCUSSION THREAD

**Discussion by jbeyermonday** [vercel/next.js#32672] [Help]
Hello!

We want to use multiple nextjs instances for production in a failsafe and scalable way and we're not sure what is the correct way to guarantee to always server the most up to date generated pages.

**Background**
- we mainly serve pages including content from a headless CMS
- we generate a few pages statically on build and rely on incremental regeneration (SSG + ISR)
- we intent to run multiple nextjs instances for load-balancing and fluent deployment

**Problem**
So we're wondering what is the envisaged way of running multiple nextjs instances, especially when it comes to caching and "stale while revalidate". 
We expect complications when, say one nextjs-instance (A) re-generates a page after request #A1 and servers it on request #A2, a third request #B3 could still reach a second nextjs-instance (B) later and serve a stale version of a page.
Is there an official way to share the "incrementally static regenerated" pages between the nextjs-instances?

**Questions we have:**
- what .next folders contents can and should be shared between instances _at runtime_? (images, pages, static, ...)
- is incremental regeneration and, by extension, static generation not suited for this use-case? (regarding SWR)
- should all static regeneration be blocking? (basically SSR)

I hope I didn't overlook anything. And yes I found discussions, many of them unanswered, talking about ways how it "might" work, disabling in-memory-cache, syncing the .next folders and similar. But we're curious about a solution that not just works "coincidentally" but in a intended way. That way we ensure upgradability in the future. Sadly, most of the official documentation seems to be favoring deployment via vercel, which is no option for us. 

Thank you.

**Comment by dennstack** [comment]
Same question here!
